fiftyone/core/map/process.py

```diff
@@ -264,3 +266,14 @@ def _map_batch(args: Tuple[int, int, fomb.SampleBatch]):
         if process_batch_count is not None:
             with process_batch_count.get_lock():
                 process_batch_count.value += 1
+
+
+def _process_worker_error(
+    process_skip_failures, process_cancel_event, process_queue, sample_id, err
+):
+    if process_skip_failures:
+        # Cancel other workers as soon as possible.
+        process_cancel_event.set()
+
+    # Add sample ID and error to the queue.
+    process_queue.put((sample_id, err, None))
```

Comment:
No need for a function here if it's not re-used.

Reply:
I keep it because it makes the code a little bit cleaner and easier to read :)

Reply:
I would disagree, I think it makes it harder to read. Now I have to bounce around to another method to see what is happening. More methods != greater readability. Especially when the content is only like 3 lines and it isn't reused.

I like to imagine coming back after 6 months without touching this code. Then being able to understand what a function is doing without have bounce around to different functions is ideal. While that's not always achievable in this case it is.

Reply:
This is bigger philosophical pattern I'm digging into. 

Is this really a big deal?
With this 1 specific method: No, not necessarily. I agree, it's still readable enough.
With 10 methods like this: Yes it makes everything far harder to understand what the single worker should being doing.

Readability should be targeted not for us, making this feature, but for developers that have never seen this code at all. My point being is let's apply good patterns globally and try not not make exceptions. Functions should only be created with good reason (re-useability, easier testing, etc.). Arbitrarily making functions because it's pretty just makes the code needlessly complicated.

Reply:
I think these are all good points; personally, I do like functions with smaller bodies, but we can move forward with this change being reverted.